tasks/src/main/scala/dagr/tasks/misc/BwaMemStreamed.scala

```diff
@@ -53,48 +54,25 @@ class BwaMemStreamed(unmappedBam: PathToBam,
                      mergeBamAlignmentMem: String = "2G",
                      fifoBufferMem: String = "512M",
                      processAltMappings : Boolean = true
-                    ) extends PipedTask with VariableResources {
-  this.name = "BwaMemStreamed"
-  override protected val tasks = ListBuffer[ProcessTask]()
+                    ) extends Pipeline {
 
-  tasks ++= SamToFastq(in = unmappedBam, out = Io.StdOut).requires(memory = Memory(samToFastqMem)).getTasks
-  tasks ++= new FifoBuffer().requires(Cores(0), Memory(fifoBufferMem)).getTasks
-  tasks ++= new BwaMem(fastq = Io.StdIn, ref = ref, minThreads = minThreads, maxThreads = maxThreads, memory = Memory(bwaMemMemory)).getTasks
-  tasks ++= new FifoBuffer().requires(memory = Memory(fifoBufferMem)).getTasks
+  override def build(): Unit = {
+    val samToFastq = SamToFastq(in=unmappedBam, out=Io.StdOut).requires(Cores(0.5), Memory(samToFastqMem))
+    val fqBuffer   = new FifoBuffer[Fastq].requires(memory=Memory(fifoBufferMem))
+    val bwaMem     = new BwaMem(fastq = Io.StdIn, ref = ref, minThreads = minThreads, maxThreads = maxThreads, memory = Memory(bwaMemMemory))
+    val bwaBuffer  = new FifoBuffer[Sam].requires(memory=Memory(fifoBufferMem))
 
-  // Optionally tie in alt-processing if requested and the .alt file exists
-  {
+    var pipe = samToFastq | fqBuffer | bwaMem | bwaBuffer
+
+    // Optionally tie in alt-processing if requested and the .alt file exists
     val alt = PathUtil.pathTo(ref + ".alt")
     if (processAltMappings && alt.toFile.exists()) {
-      tasks ++= new BwaK8AltProcessor(altFile = alt).getTasks
-      tasks ++= new FifoBuffer().requires(memory = Memory(fifoBufferMem)).getTasks
+      val k8       = new BwaK8AltProcessor(altFile = alt)
+      val k8Buffer = new FifoBuffer[Sam].requires(memory=Memory(fifoBufferMem))
+      pipe = pipe | k8 | k8Buffer
     }
```

Comment:
Thanks!  I really like this idea, and was in fact struggling with how to write this section.  I'll give this a shot.  (We could also do this for ==> instead of using NoOpTask everywhere btw).